### PR TITLE
DEVOPS-5382: Fix golang-lint tests that need to be run on entire filesystem

### PR DESCRIPTION
**Pull request by greenkiwi**
**Description**

Run on whole workspace

Fixes [#DEVOPS-5382](https://team-turo.atlassian.net//browse/DEVOPS-5382)

**Changes**

* fix(golangci-lint): run golangci-lint on whole workspace

🚀 PR created with [fotingo](https://github.com/tagoro9/fotingo)

### CODE REVIEW

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -77,6 +77,7 @@
   entry: hooks/golang/golangci-lint.sh
   language: script
   files: \.go$
+  pass_filenames: false
```
Comment (greenkiwi) [inline .pre-commit-hooks.yaml:80]: @shakefu this is the key change, we might want to add this to some of the other checks too.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
```
Comment (tagoro9) [inline .github/workflows/ci.yaml:12]: Maybe better to run on our runners

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    env:
```
Comment (tagoro9) [inline .github/workflows/ci.yaml:13]: Probably none of these are needed

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+name: Release
+# Only release when we merge to main
+on:
+  push:
+    branches: [main]
+# Handle release versioning automatically with semantic release
+jobs:
+  semantic-release:
+    name: Semantic Release
+    runs-on: ubuntu-latest
```
Comment (tagoro9) [inline .github/workflows/release.yml:10]: Run on our runners

File: .pre-commit-hooks.yaml (hunk 1 above)
Comment (tagoro9) [reply .pre-commit-hooks.yaml:80]: Maybe with this change, the hook that I added below may no longer be needed, which is running for all the go modules in the repo and was probably trying to fix some of the same issues you run into

Comment (tagoro9) [reply .pre-commit-hooks.yaml:80]: If you look at https://github.com/turo/pre-commit-hooks/blob/main/hooks/golang/golangci-lint-mod.sh the end result is very similar so probably it woudl be good to consolidate into just one hook

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (greenkiwi) [reply .github/workflows/ci.yaml:13]: I'll remove them... they will be blank anyway...

**Pull request merged**